moj_utils/views.py

```diff
@@ -0,0 +1,40 @@
+import os
+
+from django.core.exceptions import ImproperlyConfigured
+from django.http.response import JsonResponse
+from django.views.generic import View
+
+
+class PingJsonView(View):
+    """
+    View for returning IRaT information from environment variables
+    c.f. https://github.com/ministryofjustice/incident-response-schemas/blob/master/ping-dot-json-schema.json
+
+    Usage:
+        urlpatterns = [
+            ...
+            url(r'^ping.json$', PingJsonView.as_view(**environment_key_names), name='ping_json'),
+            ...
+        ]
+    """
+    build_date_key = None      # Docker container build date
+    commit_id_key = None       # Git SHA from which the image was built
+    version_number_key = None  # Docker container tag
+    build_tag_key = None       # Jenkins job name and id
+
+    def __init__(self, **kwargs):
+        if 'build_date_key' not in kwargs or 'commit_id_key' not in kwargs:
+            raise ImproperlyConfigured('ping.json schema requires build_date_key and '
+                                       'commit_id_key to be provided')
```

Comment:
the IRaT spec says `build_date` and `commit_id` are required, so i'm just
trying to force the view config for these keys to be defined. once the app
is running and the actual environment variables are missing, i think the
view should still provide a response, but it then has a status of 501 NOT
IMPLEMENTED (couldn't find a more suitable one)

On 8 December 2015 at 16:22, Ian Brechin <notifications@github.com> wrote:

> In moj_utils/views.py
> <https://github.com/ministryofjustice/django-utils/pull/3#discussion_r46975034>
> :
>
> > +    Usage:
> > +        urlpatterns = [
> > +            ...
> > +            url(r'^ping.json$', PingJsonView.as_view(**environment_key_names), name='ping_json'),
> > +            ...
> > +        ]
> > +    """
> > +    build_date_key = None      # Docker container build date
> > +    commit_id_key = None       # Git SHA from which the image was built
> > +    version_number_key = None  # Docker container tag
> > +    build_tag_key = None       # Jenkins job name and id
> > +
> > +    def __init__(self, **kwargs):
> > +        if 'build_date_key' not in kwargs or 'commit_id_key' not in kwargs:
> > +            raise ImproperlyConfigured('ping.json schema requires build_date_key and '
> > +                                       'commit_id_key to be provided')
>
> So you want them to be provided but it's ok if they are none or blank? I
> mean I can tell that it is, because you handle those cases in the get()
> method, but in that case why require that they be provided? Is it just a
> kind of hint thing?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ministryofjustice/django-utils/pull/3/files#r46975034>
> .
>